File: code/python/ch07/practice/warmup_03_sum_of_digits.py
"""
Warmup 3: Sum of Digits
==============================
Chapter 7: Number Wizardry — Math for Programmers

PROBLEM
-------
Given an integer n, return the sum of its digits.
Use the absolute value of n (ignore the sign).

INPUT FORMAT
------------
A single integer n.

OUTPUT FORMAT
-------------
A single integer: the sum of the digits of |n|.

CONSTRAINTS
-----------
- -10^9 <= n <= 10^9

EXAMPLES
--------
Input:
  12345
Output: 15

Input:
  0
Output: 0

Input:
  -456
Output: 15

Input:
  999
Output: 27

INSTRUCTIONS
------------
Replace the `pass` in the solve() function with your solution.
The main block at the bottom handles input/output — don't change it.
"""


def solve(n: int) -> int:
    """Return the sum of digits of n."""
    pass  # TODO: Replace this with your solution


# ── Do not change anything below this line ──────────────────────────
if __name__ == "__main__":
    n = int(input().strip())
    print(solve(n))

File: code/java/ch07/practice/Warmup03SumOfDigits.java
package ch07.practice;

import java.util.*;

/**
 * Warmup 03: Sum of Digits
 * ==============================
 * Chapter 7: Number Wizardry
 *
 * PROBLEM: Given integer n, return the sum of its digits. Use abs(n).
 *
 * EXAMPLES:
 *   solve(12345) = 15   (1+2+3+4+5)
 *   solve(0)     = 0
 *   solve(-456)  = 15   (4+5+6)
 *   solve(999)   = 27
 *
 * CONSTRAINTS:
 *   -10^18 <= n <= 10^18
 *
 * INSTRUCTIONS: Replace "return 0;" in solve() with your solution.
 * Do NOT convert to String!
 */
public class Warmup03SumOfDigits {
    public static int solve(long n) {
        // TODO: Replace this with your solution
        return 0;
    }

    // ── Do not change anything below this line ──────────────────────
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long n = Long.parseLong(sc.nextLine().trim());
        System.out.println(solve(n));
        sc.close();
    }
}
